lib/vauth/vauth.h

```diff
@@ -26,6 +26,15 @@
 
 struct Curl_easy;
 
+/* Generic auth data buffer support. */
+struct authbuffer {
+  void (*freefunc)(void *);
+  const unsigned char *data;
+  size_t len;
+};
+
+typedef struct authbuffer authbuffer;
```

Comment:
Please avoid custom typedefs like this and use `struct authbuffer` in code instead.

Do you think it would make sense to use a struct dynbuf here instead of data + len?

Reply:
> Please avoid custom typedefs like this 

I'll change it.

> Do you think it would make sense to use a struct dynbuf

No. I've already considered it, and gave up: the goal is not to support realloc, but a fixed data size and an associated free procedure.
Merging both notions would probably result in something very generic and useful, but is a picky design work.

Reply:
typedef now removed.